gensrc/script/vectorized/vectorized_functions.py

```diff
@@ -634,4 +634,7 @@
     [150139, 'reverse', 'ARRAY_DECIMALV2', ['ARRAY_DECIMALV2'], 'ArrayFunctions::array_reverse_decimalv2'],
     [150140, 'reverse', 'ARRAY_DATETIME',  ['ARRAY_DATETIME'],  'ArrayFunctions::array_reverse_datetime'],
     [150141, 'reverse', 'ARRAY_DATE',      ['ARRAY_DATE'],      'ArrayFunctions::array_reverse_date'],
+
+    [150150, 'array_join', 'VARCHAR', ['ARRAY_VARCHAR', 'VARCHAR'],   'ArrayFunctions::array_join_varchar'],
+    [150151, 'array_join', 'VARCHAR', ['ARRAY_VARCHAR', 'VARCHAR', 'VARCHAR'],   'ArrayFunctions::array_join_varchar'],
```

Comment:
What‘s the behavior if the input type is ARRAY_INT？ FE will directly throw an exception ?

Reply:
will first cast array_int to array_varchar
```
mysql> explain verbose select array_join(b, ',_', '*/') from m5;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Explain String                                                                                                                                                                          |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| PLAN FRAGMENT 0(F00)                                                                                                                                                                    |
|   Output Exprs:3: array_join                                                                                                                                                            |
|   Input Partition: RANDOM                                                                                                                                                               |
|   RESULT SINK                                                                                                                                                                           |
|                                                                                                                                                                                         |
|   1:Project                                                                                                                                                                             |
|   |  output columns:                                                                                                                                                                    |
|   |  3 <-> array_join[(cast([2: b, ARRAY<DATE>, true] as ARRAY<VARCHAR>), ',_', '*/'); args: INVALID_TYPE,VARCHAR,VARCHAR; result: VARCHAR; args nullable: true; result nullable: true] |
|   |  cardinality: 2                                                                                                                                                                     |
|   |                                                                                                                                                                                     |
|   0:OlapScanNode                                                                                                                                                                        |
|      table: m5, rollup: m5                                                                                                                                                              |
|      preAggregation: on                                                                                                                                                                 |
|      partitionsRatio=1/1, tabletsRatio=1/1                                                                                                                                              |
|      tabletList=252065                                                                                                                                                                  |
|      actualRows=2, avgRowSize=2.0                                                                                                                                                       |
|      cardinality: 2                                                                                                                                                                     |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
17 rows in set (0.00 sec)
```